init_friendbot.go

```diff
@@ -18,25 +19,37 @@ func initFriendbot(
 	friendbotSecret string,
 	networkPassphrase string,
 	horizonURL string,
+	rpcURL string,
 	startingBalance string,
 	numMinions int,
 	baseFee int64,
 	minionBatchSize int,
 	submitTxRetriesAllowed int,
 ) (*internal.Bot, error) {
-	if friendbotSecret == "" || networkPassphrase == "" || horizonURL == "" || startingBalance == "" || numMinions < 0 {
+	if friendbotSecret == "" || networkPassphrase == "" || startingBalance == "" || numMinions < 0 {
 		return nil, errors.New("invalid input param(s)")
 	}
+	if horizonURL == "" && rpcURL == "" {
+		return nil, errors.New("either horizon_url or rpc_url must be provided")
+	}
+	if horizonURL != "" && rpcURL != "" {
+		return nil, errors.New("only one of horizon_url or rpc_url should be provided, not both")
+	}
```

Comment:
just a suggestion, arguably harder to read

```suggestion
	if (horizonURL == "") == (rpcURL == "") {
		return nil, errors.New("exactly one of horizon_url xor rpc_url must be provided")
	}
```

Reply:
I think I'll leave it as the slightly longer code, since lines are cheap and the longer code is easier to understand (for me at least) at a glance.